### DISCUSSION TITLE
Trouble collecting stats

### DISCUSSION THREAD

**Discussion by anishshriram** [mlcommons/GaNDLF#333] [General]
**GaNDLF Version**
0.0.14-dev

**Desktop (please complete the following information):**
 - OS: macOS Monterey
 - Version: 12.2.1

**How did you install GaNDLF**
Followed instructions on documentation website:
```
git clone https://github.com/CBICA/GaNDLF.git
cd GaNDLF
conda create -n venv_gandlf python=3.7 -y
source activate venv_gandlf
conda install -c conda-forge mamba -y
mamba install -c pytorch pytorch torchvision -y
pip install -e .
```

**Dataset description**
I am using the BraTS 2021 dataset.

Description from website:
Routine clinically-acquired multi-parametric MRI (mpMRI) scans of glioma, with pathologically confirmed diagnosis and available MGMT promoter methylation status (for the glioblastoma cases with such associated data). All BraTS mpMRI scans are available as NIfTI files (.nii.gz) for Segmentation, which is the use in this issue. These mpMRI scans describe a) native (T1) and b) post-contrast T1-weighted (T1Gd), c) T2-weighted (T2), and d) T2 Fluid Attenuated Inversion Recovery (T2-FLAIR) volumes.

**Describe your question/problem**
I have finished testing and validation, with my models saved at:
```/cbica/home/shrirama/comp_space/project/meningioma/models_2```
It finished without errors.

I am trying to collect statistics from the final models, but the following error is presented.

```
Traceback (most recent call last):
  File "gandlf_collectStats", line 129, in <module>
    main()
  File "gandlf_collectStats", line 48, in main
    inputDir = os.path.normpath(args.inputDir)
AttributeError: 'Namespace' object has no attribute 'inputDir'
```

I followed the instructions on the docs website to collect stats:
```
source activate venv_gandlf
python gandlf_collectStats \
> -m /cbica/home/shrirama/comp_space/project/meningioma/models_2 \
> -o /cbica/home/shrirama/comp_space/project/meningioma/stats_2
```

Am I doing something wrong when calling `gandlf_collectStats`?

**Comment by sarthakpati** [comment]
What is L48 for `gandlf_collectStats`? Is it the same as https://github.com/CBICA/GaNDLF/blob/master/gandlf_collectStats#L48?
```python
inputDir = os.path.normpath(args.modelDir)
```
You might be running an older version of GaNDLF, perhaps?

**Reply by anishshriram** [reply]
I re-installed GaNDLF, you were correct in the L48 being different, but running it with the re-installed GaNDLF gives me the error:

```
Traceback (most recent call last):
  File "gandlf_collectStats", line 129, in <module>
    main()
  File "gandlf_collectStats", line 48, in main
    inputDir = os.path.normpath(args.modelDir)
AttributeError: 'Namespace' object has no attribute 'modelDir'
```

**Reply by sarthakpati** [reply]
Got it. You can just change this line to the following:
```python
inputDir = os.path.normpath(args.modeldir)
```
And it should work. I have put this in the PR #335

**Reply by anishshriram** [reply]
That worked – I know it is running, but the problem is both the data CSV and the plot PNG are coming up empty. Do you know why this might be?

Unrelated, but every time there are updates to gandlf, do I have to reinstall every time, or is there a way I can sync to the latest version?

**Reply by sarthakpati** [reply]
> both the data CSV and the plot PNG are coming up empty

Unsure why that's happening. You might need to debug this yourself.

> Unrelated, but every time there are updates to gandlf, do I have to reinstall every time, or is there a way I can sync to the latest version?

Yes, you can use a git clone and perform installation using `pip install -e .`

**Reply by anishshriram** [reply]
Alright thank you so much for your help.
